packages/atomic/src/components.d.ts

```diff
@@ -613,6 +613,18 @@ export namespace Components {
          */
         "store"?: AtomicInsightStore;
     }
+    interface AtomicInsightResultAction {
+        /**
+          * Specify the result action icon to display.
+         */
+        "icon": string;
+        /**
+          * The text tooltip to show on the result action icon
+         */
+        "label"?: string;
```

Comment:
Sometimes result actions are toggle buttons. Did you consider supporting different labels depending on the button state? In addition, the operation performed when activating a toggle button is not instant. We need to be able to support a loading/processing state as well.